source/ProcessManager.Core/Domain/OrchestrationInstance/OrchestrationInstance.cs

```
@@ -31,12 +31,14 @@ private OrchestrationInstance(
         OrchestrationDescriptionId orchestrationDescriptionId,
         OperatingIdentity identity,
         IClock clock,
-        Instant? runAt = default)
+        Instant? runAt = default,
+        IdempotencyKey? idempotencyKey = default)
     {
         Id = new OrchestrationInstanceId(Guid.NewGuid());
         Lifecycle = new OrchestrationInstanceLifecycle(identity, clock, runAt);
         ParameterValue = new();
         CustomState = new OrchestrationInstanceCustomState(string.Empty);
+        IdempotencyKey = idempotencyKey;
 
```

We could throw an exception if the idempotency key is too long